### PR TITLE
Save path based on platform

### PR DESCRIPTION
**Pull request by chaoskagami**
This is a very minor tweak. Your save editor is awesome, by the way. I made a patch that uses #define to allow other platforms' compilers to use proper locations for finding save files. I believe the OSX path is correct; I don't own a mac, but with qt on mac theoretically a version could be built from source so they won't be left out of the party. As for linux...there's no official version _yet_. I posted a hackish method on /r/underminers to do so (still hoping for an official release.)

I can't verify if the mac code works (it should), but the linux bit works as intended and there's no change to the windows bit.

### CODE REVIEW

**Comment by Cofeiini** [comment]
> NO!!!
> A GENUINE COMPLIMENT...!!!
> HOWEVER...
> YOU DON'T TRULY UNDERSTAND THE HIDDEN POWER OF THIS CODE!!!
> THEREFORE...
> WHAT YOU JUST SAID IS INVALID!!!
> THIS DATE WON'T ESCALATE ANY FURTHER!!!
> ... UNLESS YOU FIND MY SECRET!!
> BUT THAT WON'T HAPPEN!!

Seriously, I appreciate the help.  
I  did plan on supporting other platforms, but I wanted to have this program to work on at least one before moving on to others.  
I'll merge your changes right away. Your solution is much smarter than what I originally did.

**Comment by chaoskagami** [comment]
SPAGHETTI?

Joking aside, I just kinda patched it to get it to work here. Figured I'd ship what I did upstream. For all the dissing preprocessor macros get from Java and C++ coders, they do the job well for platform-specific code.

Now, I still have to have a bad time on Linux for testing purposes, so I'll just excuse myself while I generate a save for that. ;P

**Comment by Cofeiini** [comment]
Well, no matter how big or small, I still greatly appreciate the contribution.  
I'm not familiar with the "macro dissing", even though I've had about a year of experience in coding embedded microchips. We used old C code for that, because the chips had very limited memory.  
About that Linux testing. How familiar are you with it? Unless there's some Linux compatible version of the game, the only way to run it is with an emulated operating system or through Wine. Either way, There's going to be a Windows environment, which doesn't differ from the current one.  
And yes, my secret is...

> "SPAGHETTI."
> THAT'S WHAT YOU'RE THINKING, ISN'T IT? 
> RIGHT! 
> BUT OH-SO WRONG! 
> THIS AIN'T ANY PLAIN OL' PASTA!

**Comment by chaoskagami** [comment]
> Eat
> The taste is...indescribable.

If you're an embedded dev, then that's good. A few people I know who are C++ or Java programmers seem to be going in a very wrong direction, believing every check should be at runtime. This is also perpetuating quite a bit in places like ycombinator and the programming subreddits. I guess they seem to think preprocessor macros are the new goto or something (which has it's uses, IMO) Just kind of rambling here, sorry.

As for linux, very familiar. There is a way to run it on Linux natively, but not officially. See my post here: https://www.reddit.com/r/Underminers/comments/3xfbfn/attention_linux_users_native_is_possible/

To summarize: we grab the linux binaries from a game maker studio install and use those to run the game. ~/.config is where configuration data ends up on Linux anyways (XDG standard) so if an official version is ever released save data will end up in the same location as my hack approach. So it's partially supporting a hack, but if an official version is released no changes are needed. There's also no wine involved.

I wish it were official of course. But no.

**Comment by Cofeiini** [comment]
> WHAT A PASSIONATE EXPRESSION!!!
> YOU MUST REALLY LOVE MY CODING!
> AND BY EXTENSION, ME!!!
> MAYBE EVEN MORE THAN I DO!!!

That is strange, since Game Maker can export to Linux. There's even Ubuntu logo on their [page](https://www.yoyogames.com/studio/multiformat/desktop).  
I'm also very familiar with different Linux Distros. Currently I'm using Ubuntu Gnome, but I have experimented with every mayor distribution. Ubuntu happens to have biggest community and widest support range.  
Maybe, because I worked with embedded firmwares only, these preprocessor discussions never surfaced around me before.
